internal/audit/cluster_checks.go

```diff
@@ -214,3 +218,58 @@ func makeCheckClusterHighHAAssets(haAssetsThreshold int) checkFunc {
 		return Pass, nil
 	}
 }
+
+func checkClusterNamesForWhitespace(reader *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+	for _, clusterName := range reader.GetClusterNames() {
+		if strings.ContainsFunc(clusterName, unicode.IsSpace) {
+			examples.add("Cluster name '%s' contains whitespace", clusterName)
+		}
+	}
+
+	if examples.Count() > 0 {
+		logCritical("Found %d clusters with names containing spaces", examples.Count())
+		return Fail, nil
+	}
+
+	return Pass, nil
+}
+
+func checkLeafnodeServerNamesForWhitespace(r *archive.Reader, examples *ExamplesCollection) (Outcome, error) {
+
+	for _, clusterName := range r.GetClusterNames() {
+		clusterTag := archive.TagCluster(clusterName)
+
+		invalidLeafnodes := map[string]struct{}{}
+
+		for _, serverName := range r.GetClusterServerNames(clusterName) {
+			serverTag := archive.TagServer(serverName)
+
+			var serverLeafz server.Leafz
+			err := r.Load(&serverLeafz, clusterTag, serverTag, archive.TagServerLeafs())
+			if err != nil {
+				logWarning("Artifact 'LEAFZ' is missing for server %s", serverName)
+				continue
+			}
+
+			// check if leafnode name contains any spaces
+			for _, leaf := range serverLeafz.Leafs {
+				// check if leaf name contains any spaces
+				if strings.ContainsFunc(leaf.Name, unicode.IsSpace) {
+					invalidLeafnodes[leaf.Name] = struct{}{}
+				}
+			}
+		}
+
+		if len(invalidLeafnodes) > 0 {
+			examples.add("Detected leafnodes with whitespaces in cluster %s: %v", clusterName, maps.Keys(invalidLeafnodes))
```

Comment:
Same here, this is an example, not a printf. So maybe:
  `"Cluster %s: %v", clusterName, map.Keys`